src/coreclr/src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -54,28 +54,26 @@ public enum GCNotificationStatus
     public static class GC
     {
         [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern void GetMemoryInfo(out ulong highMemLoadThresholdBytes,
-                                                  out ulong totalAvailableMemoryBytes,
-                                                  out ulong lastRecordedMemLoadBytes,
-                                                  out uint lastRecordedMemLoadPct,
-                                                  // The next two are size_t
-                                                  out UIntPtr lastRecordedHeapSizeBytes,
-                                                  out UIntPtr lastRecordedFragmentationBytes);
-
-        public static GCMemoryInfo GetGCMemoryInfo()
-        {
-            GetMemoryInfo(out ulong highMemLoadThresholdBytes,
-                          out ulong totalAvailableMemoryBytes,
-                          out ulong lastRecordedMemLoadBytes,
-                          out uint _,
-                          out UIntPtr lastRecordedHeapSizeBytes,
-                          out UIntPtr lastRecordedFragmentationBytes);
-
-            return new GCMemoryInfo(highMemoryLoadThresholdBytes: (long)highMemLoadThresholdBytes,
-                                    memoryLoadBytes: (long)lastRecordedMemLoadBytes,
-                                    totalAvailableMemoryBytes: (long)totalAvailableMemoryBytes,
-                                    heapSizeBytes: (long)(ulong)lastRecordedHeapSizeBytes,
-                                    fragmentedBytes: (long)(ulong)lastRecordedFragmentationBytes);
+        private static extern void GetMemoryInfo(GCMemoryInfoData data, int kind);
+
+        public static GCMemoryInfo GetGCMemoryInfo() => GetGCMemoryInfo(GCKind.Any);
+
+        public static GCMemoryInfo GetGCMemoryInfo(GCKind kind)
+        {
+            if ((kind >= GCKind.Any) && (kind <= GCKind.Background))
```

Comment:
The usual style for error handling is to put it first, e.g.:

```
if ((kind <  GCKind.Any) || (kind > GCKind.Background))
    throw new ...

var data = new GCMemoryInfoData();
GetMemoryInfo(data, (int)kind);
return new GCMemoryInfo(data);
```

Reply:
fixed